web/vibe/web/internal/rest/jsclient.d

```diff
@@ -100,12 +105,24 @@ class JSRestClientSettings
 			// extract the server part of the URL
 			auto burl = URL(intf.baseURL);
 			burl.pathString = "/";
-			fout.serializeToJson(burl.toString()[0 .. $-1]);
-			// and then assemble the full path piece-wise
-			foreach (p; route.fullPathParts) {
-				fout.put(" + ");
-				if (!p.isParameter) fout.serializeToJson(p.text);
-				else fout.formattedWrite("toRestString(%s)", p.text);
+
+			// if url not empty
+			if (intf.baseURL != intf.basePath) {
```

Comment:
I think using `intf.baseURL != URL.init` would make this a bit more obvious here.

Reply:
`intf.baseURL` is a `string`, not `URL`. `URL.init` is empty, `baseURL == basePath` can be if `URL.init` is passed to `RestInterface.ctor` [here](https://github.com/vibe-d/vibe.d/blob/master/web/vibe/web/internal/rest/common.d#L41)
I don't understand why `RestInterface` use `strings` for `baseURL` and `basePath` instead `URL` for both. Is it principled? Why `URL` can't be initialize with empty string? Why `URL.init` prints not empty string? Maybe it can be changed? I think it can simplify code.